### ISSUE TITLE
Crash with 224 vehicles available in scenario

### ISSUE THREAD

**Issue by luk3ZPL** [OpenLoco/OpenLoco#1376]
**Version information:**
 - Platform: W7
 - Version: 22.03.1
 - Build/commit: a759b81

**Describe the bug**
The bug is ugly it crashes the game.

**To Reproduce**
Steps to reproduce the behaviour:
0. Start the game in windows mode and try to resize it manually little bit
1. Start scenario with 224 vehicles (you may create it in an editor)
2. Click on Build trains then open aircrafts tab (or any other)
3. Try to go to the main menu
4. Catch the error

**Expected behaviour**
Go to the main menu without crash the game.

**Screenshots**
https://www.mediafire.com/view/icdio02uxn8d2tb/crash_wow.gif/file

**Additional context**
Only in scenario with 224 vehicles available.
Dump file: https://www.mediafire.com/file/iibarv2kxhatwh3/ace29f8b-8c6d-484c-86e1-8418dd3101b8%2528OpenLoco%252C_v22.03-3_%2528a759b81%2529%2529.7z/file

**Comment by LeftofZen** [comment]
@luk3ZPL are you able to upload the scenario here in a zip file?

**Comment by luk3ZPL** [comment]
> 

Yeap. Done.

**Comment by LeftofZen** [comment]
Thanks, even just loading the scenario up in scenario editor caused the game to freak out and quick back to menu for some reason.

**Comment by LeftofZen** [comment]
I also tried making a scenario with the max objects in each category and that causes the game to crash before it gets to landscape generation, I presume that the same or similar/related crash.

**Comment by luk3ZPL** [comment]
> Thanks, even just loading the scenario up in scenario editor caused the game to freak out and quick back to menu for some reason.

You may use my ObjData files which I used in scenario editor https://www.mediafire.com/file/q2jwve8ospdmhl3/ObjData.7z/file
Scenario loading for me working fine there is no problem, so you probably missing some objects.

**Comment by LeftofZen** [comment]
Thanks, are you able to upload the save game as well, I'm unable to open the scenario in the landscape/scenario editor :| yet another bug that'll need fixing

**Comment by luk3ZPL** [comment]
> Thanks, are you able to upload the save game as well, I'm unable to open the scenario in the landscape/scenario editor :| yet another bug that'll need fixing

Yes. But remember - first you need to load/start a scenario with 224 vehicles and then load a savegame without 224 veh. I have savegame without 224 vehicles (dunno if it helps). Maybe it is a memory related problem (game do not clean memory from previous state) ?

**Comment by duncanspumpkin** [comment]
Crashing in gameCommand::do21 which is the loadSave command

**Comment by luk3ZPL** [comment]
> Crashing in gameCommand::do21 which is the loadSave command `0x00448FE9` which is a graphics draw sprite function??

It could be. On the scrsht (screenshot) it crashed when I want to Quit to menu (resized window earlier). I can attach other dump. Give me a while. Maybe wrong function 1 caused 2 bugs.

**Comment by luk3ZPL** [comment]
Game even crashes when I make scenario in scenario editor with selected all 170+ vehicles. This could be bug with Scenario Editor. Steps to reproduce the behaviour:

0. Start the game in windows mode.
1. Start scenario which was made in scenari editor (all vehicles must be selected, also other stuff should be added like terrains or industry)
3. Click on "Build Trains" then open aircrafts tab (or any other) and do not close this window
4. Load the game with other vanilla scenario.
5. Click on "Build aircrafts" or Build any other vehicle which tab was previously selected and not closed
Catch the error

[e88d7ab2-1187-423e-805a-3db433a781bf(OpenLoco, v22.03-3 (a759b81)).zip](https://github.com/OpenLoco/OpenLoco/files/8357157/e88d7ab2-1187-423e-805a-3db433a781bf.OpenLoco.v22.03-3.a759b81.zip)

**Comment by luk3ZPL** [comment]
I've just figured it out. Vanilla objects from Scenario Editor are crashing the game.
1st object is: Flatbed Wagon weight 6t. There are more obj which also crashing as far as I can see.
When I excluded it from list of vehicles something strange happened - the file name was messed up by the game "14 rest.SC5". This is vanilla scenario & vanilla objets.
To be continued..
[14 rest.zip](https://github.com/OpenLoco/OpenLoco/files/8357305/1.4.rest.zip)
.

**Comment by LeftofZen** [comment]
Nice work, do you have any ideas how to fix the problem?

**Comment by luk3ZPL** [comment]
> Nice work, do you have any ideas how to fix the problem?
I think C.S. (did add) added smaller wagoons later during development and forgot some variables or implement it wrongly 
(he definitely forgot something to add or change eg. it could be bad pointer/array/buffer overflow). Current solusion could be forbid those vehicles from selecting in scenario editor. I don't know how much affected vanilla scenarios are by this bug.

https://www.tt-wiki.net/wiki/Variable:List#Class_17_-_Vehicles

**Comment by luk3ZPL** [comment]
I have 1 bad and 1 good news today. Good news is that vanilla Locomotion is not affected by this bug and bad news is that OpenLoco is affected. It is not C.S. fault during game development. Something wrong is with OpenLoco.

**Comment by LeftofZen** [comment]
Is that list on tt-wiki for TTD or Loco?

**Comment by luk3ZPL** [comment]
> Is that list on tt-wiki for TTD or Loco?

Loks like Loco https://www.tt-wiki.net/wiki/Category:Locomotion_development
Anyway I've also checked this bug crashes the game since [v20.05](https://github.com/OpenLoco/OpenLoco/releases/tag/v20.05).
